charts/super-agent/templates/uninstall-job.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if .Values.helm.create -}}
+{{- $installJobName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "uninstall-job" ) -}}
+{{- $configMapName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "manifests" ) -}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-weight: "10000"
```

Comment:
I do not think the weight is needed, if it is the same Helm should still respect the classical order

Reply:
Done on [#b5da77e](https://github.com/newrelic/helm-charts/pull/1175/commits/b5da77e706e5cce2c1b77e36f72063bd23c1e715)

Reply:
I reopen it because you might have a different view of this.

I removed it on the commit above and it started to fail all the tests.
![Screenshot 2023-10-25 at 16 50 12](https://github.com/newrelic/helm-charts/assets/53659978/8fde136d-b651-4274-96b4-05cabcc1fddd)

I readded the weight on the t commit ([#e7cfe43](https://github.com/newrelic/helm-charts/pull/1175/commits/e7cfe435e02f6e76258e114a9bff65fd20d2406b)) and it started to pass the tests again.

I think there is another race condition because of hook defaults.

[Deletion policies](https://helm.sh/docs/topics/charts_hooks/#hook-deletion-policies) default to `before-hook-creation` so the job and the configMap would keep dangling on the cluster and possibly leaking data as the values are contained on the `HelmRelease` object, hence my policy of delete also on `hook-succeeded`.

But it seems that `hook-succeeded` creates the configMap and deletes it as fast as it is created if you do not set a `weight`.

I am still investigating this tho. Let me know what you think.